eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/EventReceiverTest.java

```diff
@@ -0,0 +1,375 @@
+            new EventReceiverOptions().consumerGroup(getConsumerGroupName()));
+        final Flux<EventData> enqueuedTimeData = receiver.receive();
+
+        // Verification
+        StepVerifier.create(enqueuedTimeData.take(NUM_OF_EVENTS))
+            .expectNextCount(NUM_OF_EVENTS)
+            .verifyComplete();
+        final EventData enqueuedTimeEvent = enqueuedTimeData.toIterable().iterator().next();
+
+        // Arrange: EventPosition.fromSequenceNumber
+        receiver = client.createReceiver(PARTITION_ID, EventPosition.fromSequenceNumber(enqueuedTimeEvent.sequenceNumber(), false),
+            new EventReceiverOptions().consumerGroup(getConsumerGroupName()));
+        final Flux<EventData> sequenceNumData = receiver.receive();
+
+        // Verification
+        StepVerifier.create(sequenceNumData.take(NUM_OF_EVENTS))
+            .expectNextCount(NUM_OF_EVENTS)
+            .verifyComplete();
+        final EventData sequenceNumEvent = sequenceNumData.toIterable().iterator().next();
+
+        // Assertion
+        Assert.assertEquals(sequenceNumEvent.sequenceNumber(), enqueuedTimeEvent.sequenceNumber() + 1);
+    }
+
+    /**
+     * Test for received body and properties of a same event
+     */
+    @Ignore
+    @Test
+    public void testReceivedBodyAndProperties() {
+        skipIfNotRecordMode();
+        // Data
+        final EventData event = new EventData(PAYLOAD.getBytes());
+        event.properties().put(PROPERTY2, PROPERTY_VALUE2);
+        event.properties().put(PROPERTY1, PROPERTY_VALUE1);
+
+        // Arrange
+        receiver = client.createReceiver(PARTITION_ID, EventPosition.latest(),
+            new EventReceiverOptions().consumerGroup(getConsumerGroupName()));
+
+        final EventSender sender = client.createSender(senderOptions);
+        sender.send(event);
+
+        // Action and Verify
+        StepVerifier.create(receiver.receive())
+            .expectNextMatches(data -> {
+                validateReceivedEvent(data);
+                resendEventData = data;
+                return true;
+            })
+            .verifyComplete();
+
+        // Action and Verify
+        sender.send(resendEventData);
+        StepVerifier.create(receiver.receive())
+            .expectNextMatches(data -> {
+                validateReceivedEvent(data);
+                return true;
+            })
+            .verifyComplete();
+    }
+
+    private void validateReceivedEvent(EventData eventData) {
+        Objects.requireNonNull(eventData);
+        Assert.assertEquals(PAYLOAD, new String(eventData.body().array()));
+
+        final Map<String, Object> propertiesMap = eventData.properties();
+        Assert.assertTrue(propertiesMap.containsKey(PROPERTY1) && propertiesMap.get(PROPERTY1).equals(PROPERTY_VALUE1));
+        Assert.assertTrue(propertiesMap.containsKey(PROPERTY2) && propertiesMap.get(PROPERTY2).equals(PROPERTY_VALUE2));
```

Comment:
Do not chain assertions together. this makes it hard to figure out why something failed.